syncd/Syncd.cpp

```diff
@@ -4738,7 +4739,11 @@ void Syncd::run()
 
     m_mdioIpcServer->stopMdioThread();
 
-    sai_status_t status = removeAllSwitches();
+    sai_status_t status = SAI_STATUS_SUCCESS;
+    if (shutdownType != SYNCD_RESTART_TYPE_COLD || 0 != strcmp(SONIC_ASIC_PLATFORM, "mellanox"))
```

Comment:
@Junchao-Mellanox having a static string comparison looks very strange. Can we use a preprocessor to add a vendor specific logic instead?